app/screens/image_preview/downloader.ios.js

```diff
@@ -76,82 +97,149 @@ export default class Downloader extends PureComponent {
     };
 
     toggleDownloader = (show = true) => {
-        const {deviceHeight} = this.props;
+        const {deviceHeight, prompt} = this.props;
         const top = show ? (deviceHeight / 2) - 100 : deviceHeight;
 
         Animated.spring(this.state.downloaderTop, {
             toValue: top,
             tension: 8,
             friction: 5
         }).start(() => {
-            if (show) {
+            if (show && !prompt) {
                 this.startDownload();
             }
         });
     };
 
     startDownload = async () => {
+        const {file, downloadPath, prompt, saveToCameraRoll} = this.props;
+
         try {
-            const {file} = this.props;
+            if (this.state.didCancel) {
+                this.setState({didCancel: false});
+            }
+
             const imageUrl = Client4.getFileUrl(file.id);
+            const options = {
```

Comment:
Is there some way we can remove all this duplicated code between this and the FileAttachment component?